### PR TITLE
feat(plugin-utils): extra transform targets

### PR DESCRIPTION
**Pull request by harlan-zw**
This is an edge case on the Nuxt Docus package, it needs to include a css file within node_modules that has @apply's in it. Since this is an edge case and would be primarily done via hooks, I've avoided glob behaviour.

This is powerful since it lets them provide easily themeable CSS via the end-user modifying the windi config.

### CODE REVIEW

File: packages/plugin-utils/src/options.ts [hunk 1]
```diff
@@ -261,6 +261,10 @@ export interface ResolvedOptions {
     runOnStartup: boolean
     transformers: TransformerFunction[]
     extractors: Extractor[]
+    extraTransformTargets: {
+      css: string[]
+      groups: string[]
```
Comment (antfu) [inline packages/plugin-utils/src/options.ts:266]:
```suggestion
      detect: string[]
```

It's not only about groups, so maybe it's better to have a more general name

Comment (harlan-zw) [reply packages/plugin-utils/src/options.ts:266]: True, thanks

File: packages/plugin-utils/src/resolveOptions.ts [hunk 2]
```diff
@@ -99,6 +99,10 @@ export async function resolveOptions(
       runOnStartup: true,
       transformers: [],
       extractors: [],
+      extraTransformTargets: {
+        css: [],
+        groups: [],
```
Comment (hannoeru) [inline packages/plugin-utils/src/resolveOptions.ts:104]:
```suggestion
        detect: [],
```

Comment (harlan-zw) [reply packages/plugin-utils/src/resolveOptions.ts]: Thanks han, just fixed that commit in a separate commit

**Pull request merged**